client/app/private-database/order/sql-database-order-controller.js

```diff
@@ -203,6 +203,12 @@ angular.module('App').controller(
       }
     }
 
+    shouldDisplayFreeDbWarning() {
+      return this.isPrivateDb()
+        && _.get(this.model, 'ram') === '512'
```

Comment:
`this.model.ram` and `this.selectedHosting. stillHasFreeDbOffer ` can be empty at this point ?

Reply:
The method is called from the DOM in the digest cycle, so it can be called before any promise.